srv/salt/state/miniconda.sls

```diff
@@ -1,26 +1,24 @@
 # File:/srv/salt/miniconda.sls
 
-download_miniconda_installer:
+miniconda_installed:
+  cmd.run:
+    - name: source ~/miniconda3/etc/profile.d/conda.sh && conda --version
+
+miniconda_installer_present:
   file.managed:
     - name: /home/centos/Miniconda3-latest-Linux-x86_64.sh
     - source: https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
     - skip_verify: True
-    - keep_source: True  # keep to avoid re-downloading.
+    - keep_source: True  # Keep to avoid re-downloading.
     - mode: 700
-
-add_miniconda_to_bashrc:  # Add first, so that check in install_miniconda will work if conda is already installed.
-  file.replace:
-    - name: ~/.bashrc
-    - pattern: /^export PATH="/root/miniconda3/bin:$PATH"/
-    - repl: export PATH="/root/miniconda3/bin:$PATH"
-    - append_if_not_found: True
+    - onfail:
+      - cmd: miniconda_installed
 
 install_miniconda:
   cmd.run:
-#    - name: 'sudo /home/centos/Miniconda3-latest-Linux-x86_64.sh -b -u'
-#    - "conda --version || sudo /home/centos/Miniconda3-latest-Linux-x86_64.sh -b -u"
-    - name: "echo $(conda --version) && conda --version && echo 'we have conda' || echo 'we did not have conda' && sudo /home/centos/Miniconda3-latest-Linux-x86_64.sh -b -u"
-
-    - unless:
-        - cmd.run: echo $(conda --version)
+    - name: 'sudo /home/centos/Miniconda3-latest-Linux-x86_64.sh -b -u'
```

Comment:
I think -u forces a install over the miniconda directory even if it exists. it might be better to not have that flag. Let me know if I got it wrong the purpose of that flag.

Reply:
I can't actually find documentation for that `-u` flag. 
I did find [this](https://github.com/NanoComp/meep/issues/460#issuecomment-415569758) which suggests `-u` upgrades an existing installation. Whether it recognises that it is trying to install the already installed version and stops, idk.

Arguably, with the way the states are structured, if conda is being installed, it's because the checking state couldn't find conda, so either it's not there, or something is broken (whether conda itself, or the bash magic to access via `conda --version`), so forcing a reinstall via `-f` might be preferred in that case. Might have to specify the path with a `-p` flag if you want that, I haven't experimented. 

According to the [doc](https://docs.anaconda.com/anaconda/install/silent-mode/) the `-b` flag doesn't modify `~.bashrc` - which, although I think adding to the path automatically would be preferable, does allow the installer to processed without manual intervention (I tried and it hang without).